data/json/effect_on_condition.json

```diff
@@ -505,128 +505,126 @@
   {
     "type": "effect_on_condition",
     "id": "EOC_GIVE_RANDOM_MISSION",
-    "recurrence": 1,
-    "global": false,
-    "condition": { "or": [ { "npc_has_trait": "GETS_RANDOM_MISSION" }, { "npc_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
+    "condition": { "or": [ { "u_has_trait": "GETS_RANDOM_MISSION" }, { "u_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
     "effect": [
       {
         "weighted_list_eocs": [
           [
             {
               "id": "give_MISSION_GET_RELIC",
-              "condition": { "or": [ { "npc_has_trait": "GETS_RANDOM_MISSION" }, { "npc_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
+              "condition": { "or": [ { "u_has_trait": "GETS_RANDOM_MISSION" }, { "u_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
               "effect": { "offer_mission": "MISSION_GET_RELIC" }
             },
             { "const": 1 }
           ],
           [
             {
               "id": "give_MISSION_GET_RECORD_ACCOUNTING",
-              "condition": { "npc_has_trait": "GETS_RANDOM_MISSION" },
+              "condition": { "u_has_trait": "GETS_RANDOM_MISSION" },
               "effect": { "offer_mission": "MISSION_GET_RECORD_ACCOUNTING" }
             },
             { "const": 1 }
           ],
           [
             {
               "id": "give_MISSION_GET_INHALER",
-              "condition": { "or": [ { "npc_has_trait": "GETS_RANDOM_MISSION" }, { "npc_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
+              "condition": { "or": [ { "u_has_trait": "GETS_RANDOM_MISSION" }, { "u_has_trait": "GETS_RANDOM_MISSION_NOMOVE" } ] },
```

Comment:
We should add an "and" condition here to check that the NPC has the asthma trait as well.